stream_reader.go

```diff
@@ -32,41 +32,43 @@ func (stream *streamReader[T]) Recv() (response T, err error) {
 
 	var emptyMessagesCount uint
 
-waitForData:
-	line, err := stream.reader.ReadBytes('\n')
-	if err != nil {
-		respErr := stream.errAccumulator.unmarshalError()
-		if respErr != nil {
-			err = fmt.Errorf("error, %w", respErr.Error)
+	for {
+		var line []byte
+		line, err = stream.reader.ReadBytes('\n')
+		if err != nil {
+			respErr := stream.errAccumulator.unmarshalError()
+			if respErr != nil {
+				err = fmt.Errorf("error, %w", respErr.Error)
+			}
+			return
 		}
-		return
-	}
 
-	var headerData = []byte("data: ")
-	line = bytes.TrimSpace(line)
-	if !bytes.HasPrefix(line, headerData) {
-		if writeErr := stream.errAccumulator.write(line); writeErr != nil {
-			err = writeErr
-			return
+		var headerData = []byte("data: ")
+		line = bytes.TrimSpace(line)
+		if !bytes.HasPrefix(line, headerData) {
```

Comment:
Can we maybe flip this if condition?

```go
if bytes.HasPrefix(line, headerData) {
    line = bytes.TrimPrefix(line, headerData)
    // ... check for [DONE], etc.
}
```

In that case all prefix-manipulating logic is located close-by

Reply:
Yes, we can. Just as you mentioned, if we move the logic for manipulating the prefix closer to the condition statement, it would make sense.

However, I believe it's better to leave it as it is due to the [happy-path idiom](https://maelvls.dev/go-happy-line-of-sight/) and the fail-fast principle. It's a good practice to keep things organized in this way.👍

By the way, I share the same opinion about keeping related processes close to each other. I'm currently working on another pull request to improve that aspect.